File: public/examples/integrations/toolkits/box_api/get_legal_hold_policy_assignments_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "BoxApi.GetLegalHoldPolicyAssignments"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=TOOL_NAME
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'legal_hold_policy_id': 'lhp_9b3f2a7d',
    'filter_by_assignment_type': 'file_version',
    'filter_by_item_id': '1234567890',
    'pagination_marker': 'marker_abc',
    'maximum_items_per_page': 100,
    'response_fields': ['id', 'assigned_to', 'assignment_type', 'created_at']
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/box_api/get_legal_hold_policy_assignments_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "BoxApi.GetLegalHoldPolicyAssignments";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "legal_hold_policy_id": "lh_9f8a7b6c",
  "filter_by_assignment_type": "file",
  "filter_by_item_id": "1234567890",
  "pagination_marker": "marker_abc123",
  "maximum_items_per_page": 50,
  "response_fields": [
    "id",
    "assigned_at",
    "assigned_by",
    "item_type"
  ]
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
